### PR TITLE
Update CHANGELOG.md for 24.5

### PR DESCRIPTION
**Pull request by pratheekshasn**
Updating CHANGELOG.md with the latest changes made to 24.5.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -35,6 +35,41 @@ Branch: `nilrt/master/next`
 - [Fixed](https://github.com/ni/meta-nilrt/pull/676) a parsing error in the recovery media provisioning scripts.
 
 
+----
+## 10.3
+Branch: `nilrt/24.3/kirkstone`
```
Comment (amstewart) [inline CHANGELOG.md]: The branch ref for NILRT 10.3 should be `nilrt/24.5/kirkstone`. You can check the versions [here](https://dev.azure.com/ni/DevCentral/_wiki/wikis/AppCentral.wiki/27079/NILRT-Versions#).

File: CHANGELOG.md [hunk 2]
```diff
@@ -35,6 +35,41 @@ Branch: `nilrt/master/next`
...
+
+### nilrt
+#### Added
+- [Added](https://github.com/ni/nilrt/pull/269) libncurses-dev package to the required host dependencies in READEM.kernel.md.
+
+#### Changed
+- [Changed](https://github.com/ni/nilrt/pull/276) a command in README.md to rename the run-mode base image.
```
Comment (amstewart) [inline CHANGELOG.md]: We generally don't need to document changes to documentation (including READMEs) or test packages. CHANGELOG readers normally care about: (a) functional changes to how the software *works* or (b) changes to how the project is organized or packaged.

So the later comment about removing the feed-changelogs is fine to include (layout). But we can drop these content changes to the README.

File: CHANGELOG.md [hunk 3]
```diff
@@ -35,6 +35,41 @@ Branch: `nilrt/master/next`
...
+#### Removed
+- [Removed](https://github.com/ni/nilrt/pull/274) feed-changelogs, since we don't yet have a good usecase for their inclusion, and they have not proven to be useful.
+
+### meta-nilrt
+#### Added
+- [Added](https://github.com/ni/meta-nilrt/pull/690/) support for getting full disk image.
+- [Added](https://github.com/ni/meta-nilrt/pull/694) setall functionality and misc.
+- [Added](https://github.com/ni/meta-nilrt/pull/698) grub entries and improved set image handling.
+- [Added](https://github.com/ni/meta-nilrt/pull/703) support for custom images, improved error handling and UI.
+- [Added](https://github.com/ni/meta-nilrt/pull/704) compatibilty check, support for answers file, progress bar.
```
Comment (amstewart) [inline CHANGELOG.md]: Assume that CHANEGLOG readers are familiar with the high-level workflows for the software, but not their implementation details. They're going to be referencing this file when a workflow doesn't work, or to understand if we have changed some critical assumption of *how* it works. So entries like "Added setall functionality and misc." won't have much meaning to them.

Instead, you can replace this whole block with any entry about "adding a workflow to the provisioning tool to make and apply whole system images", with a link to the first PR that enabled that workflow. @chaitu236 Can help you identify which PR that is.

File: CHANGELOG.md [hunk 4]
```diff
@@ -35,6 +35,41 @@ Branch: `nilrt/master/next`
...
+
+#### Changed
+- [Changed](https://github.com/ni/meta-nilrt/pull/683) DISTRO_VERSION, NILRT_FEED_NAME in nilrt.conf.
```
Comment (amstewart) [inline CHANGELOG.md]: Likewise to the documentation and test changes, version bumps are normally not included in CHANGELOGs. Presumably, the reader already knows what CHANGELOG version they're looking at.

File: CHANGELOG.md [hunk 5]
```diff
@@ -35,6 +35,41 @@ Branch: `nilrt/master/next`
...
+- [Changed](https://github.com/ni/meta-nilrt/pull/688) test to run from a developer machine.
+- [Changed](https://github.com/ni/meta-nilrt/pull/689) test to display current manifest details in all cases.
+- [Fixed](https://github.com/ni/meta-nilrt/pull/691) run-time error caused by missing parameter.
+- [Changed](https://github.com/ni/meta-nilrt/pull/692) test to ensure recent manifest is not earlier than basis manifest.
+- [Changed](https://github.com/ni/meta-nilrt/pull/693) test to make diff hash stable despite stripped versions.
+- [Renamed](https://github.com/ni/meta-nilrt/pull/695) command line args to specify manifests.
+- [Improved](https://github.com/ni/meta-nilrt/pull/696) logging output.
+- [Changed](https://github.com/ni/meta-nilrt/pull/697) test to pass dynamic command-line arguments.
+- [Changed](https://github.com/ni/meta-nilrt/pull/702) the installation of influxdb in the Dockerfile.
```
Comment (amstewart) [inline CHANGELOG.md]: These are all testing changes that don't need to be included.

**Review by amstewart — CHANGES_REQUESTED** [review]
I think most of the changes this cycle have been either (A) the USB reimaging feature or (b) testing and documentation changes that we don't need in the CHANGELOG. So I think we can slim down this release's section quite a bit.

File: CHANGELOG.md (hunk 3 above)
Comment (chaitu236) [reply CHANGELOG.md]: https://github.com/ni/meta-nilrt/pull/694 added set image functionality at which point we had rudimentary get/set image; so we could add this PR.

**Review by chaitu236 — CHANGES_REQUESTED** [review]
^ what @amstewart said.

Also,
- please squash the two commits into one
- make sure to follow commit description convention; i.e., `<filename>: <change desc>`. E.g., refer https://github.com/ni/nilrt/commit/bc5f517f807654da5d0846b1347c02eafce88408
- add a signed off line.

File: CHANGELOG.md (hunk 1 above)
Comment (pratheekshasn) [reply CHANGELOG.md]: I have fixed this now.

File: CHANGELOG.md (hunk 2 above)
Comment (pratheekshasn) [reply CHANGELOG.md]: I have fixed this now.

File: CHANGELOG.md (hunk 3 above)
Comment (pratheekshasn) [reply CHANGELOG.md]: I have fixed this now.

File: CHANGELOG.md (hunk 5 above)
Comment (pratheekshasn) [reply CHANGELOG.md]: Removed testing changes.

**Comment by pratheekshasn** [comment]
Made all the requested changes, squashed the last two commits and added a signed-off line.

File: CHANGELOG.md (hunk 4 above)
Comment (pratheekshasn) [reply CHANGELOG.md]: I have fixed this now.

**Review by amstewart — APPROVED** [review]

**Review by chaitu236 — APPROVED** [review]

**Comment by chaitu236** [comment]
Cherry-picked to `nilrt/24.5/kirkstone`

**Pull request merged**